### DISCUSSION TITLE
The relationship between Port and Channel

### DISCUSSION THREAD

**Discussion by lisuxiaoqi** [cosmos/ibc#658] [General]
Hi All,

I'm researching the IBC those days. It's really a great idea and thanks for your work on this.

In the link https://github.com/cosmos/ibc/tree/master/spec/core/ics-005-port-allocation#motivation here comes the description:

"*Modules may bind to multiple ports and connect to multiple ports bound to by another module on a separate machine. **Any number of (uniquely identified) channels can utilise a single port simultaneously**. Channels are end-to-end between two ports, each of which must have been previously bound to by a module, which will then control that end of the channel.*"

In my understanding, a channel is a logic connection between two ports. So if there are two ports named PortA and PortB, then I can create a channel ChannelAB between PortA and PortB. After the creation of ChannelAB, PortA should be treated as occupied. It's impossible to create another channel as ChannelAC under PortA again. 

Base on my understanding, exactly only one channel can utilise a single port. It really confuses me a lot here. Could you please help to correct me? Thanks in advance!

**Comment by crodriguezvega** [comment]
Hi @lisuxiaoqi. Thanks for asking and sorry that we didn't reply earlier...

Following your example, you can actually create ChannelAC, the channel ID on the A side will be just different that the channel ID that the channelAB got when it was set up before. That's it: for example, if channelAB had the channel ID `channel-0` on the A side, then channelAC will be created with the channel ID `channel-1` on the A side. And this is what the spec reflect when saying that _any number of (uniquely identified) channels can utilise a single port simultaneously_.

I hope this helps to answer your question. We also recommend to ask general IBC questions on [Discord](https://discord.com/channels/669268347736686612/699634178173698108), so feel free to post there any other questions you may have.
